test/GUI/test_gui.py

```diff
@@ -6,133 +6,114 @@
+        mock_gui.wait_until_page_contains_element.assert_called_with("some_locator")
+        robot_call.assert_called_with(15, 1, "Set Focus To Element", "some_locator")
+        mock_gui.wait_until_element_is_visible.assert_called_with("some_locator")
 
-    @patch('robot.libraries.BuiltIn.BuiltIn.wait_until_keyword_succeeds')
-    @patch('SeleniumLibrary.ElementKeywords.click_element')
-    def test_wait_for_and_click_element_simple(self, robot_call, click_element):
+    def test_wait_for_and_click_element_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_click_element(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert click_element.called_with("some_locator")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.click_element.assert_called_with("some_locator")
 
-    @patch('robot.libraries.BuiltIn.BuiltIn.wait_until_keyword_succeeds')
-    @patch('SeleniumLibrary.FormElementKeywords.input_text')
-    def test_wait_for_and_input_text_simple(self, robot_call, input_text):
+    def test_wait_for_and_input_text_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_input_text(mock_gui, "some_locator", "text")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert input_text.called_with("some_locator", "text")
-        
-    @patch('SeleniumLibrary.WaitingKeywords.wait_until_page_contains_element')
-    @patch('SeleniumLibrary.FrameKeywords.select_frame')
-    def test_wait_for_and_select_frame_simple(self, robot_call, select_frame):
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.input_text.assert_called_with("some_locator", "text")
+
+    def test_wait_for_and_select_frame_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_select_frame(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert select_frame.called_with("some_locator")
+        mock_gui.wait_until_page_contains_element.assert_called_with("some_locator")
+        mock_gui.select_frame.assert_called_with("some_locator")
 
-    @patch('SeleniumLibrary.WaitingKeywords.wait_until_page_contains_element')
-    @patch('SeleniumLibrary.FrameKeywords.select_frame')
-    def test_unselect_and_select_frame_simple(self, select_frame, robot_call):
+    def test_unselect_and_select_frame_simple(self):
         mock_gui = Mock()
         GUILibrary.unselect_and_select_frame(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert select_frame.called_with("some_locator")
+        mock_gui.wait_for_and_select_frame.assert_called_with("some_locator")
         assert mock_gui.unselect_frame.called
```

Comment:
Should be changed to `mock_gui.unselect_frame.assert_called` I believe.

Reply:
this one is correct, I'll fix that shortly